src/fileDownloader.js

```diff
@@ -1,21 +1,38 @@
 import jsPDF from 'jspdf'
-// import(/* webpackChunkName: "jspdf", webpackPrefetch: -100  */ 'jspdf').then(({ default: jsPDF })=>{
 
-export default function download({filename, images=[] }){
+const printableWidth = 190 // 210mm (A4 paper) - 10mm margin * 2 = 190mm 
+const printableHeight = 277 // 277mm (A4 paper) - 10mm margin * 2 = 277mm
+const format = 'a4'
+  
+function getImagePrintSize(proportion){
+  const isLandscaped = proportion < 1
+  const width = isLandscaped? printableWidth / proportion: printableWidth 
+  const height = isLandscaped? printableWidth: printableWidth * proportion 
+
+  let correctionFactor = 1
+  if((isLandscaped && width>printableHeight) || (!isLandscaped && width>printableWidth)){
+    correctionFactor = (1-((width-printableWidth)/printableWidth));
+  } else if ((isLandscaped && height>printableWidth) || (!isLandscaped && height>printableHeight)){
+    correctionFactor = (1-((height-printableHeight)/printableHeight));
+  }
```

Comment:
this ensures that even on wacky situations (extremely thin and long, or extremely wide screens) the image has always the maximum width or height allowed.